build/ci.bazelrc

```diff
@@ -44,35 +44,49 @@ build:ci-limit-storage --copt="-fno-standalone-debug"
 
 # ci-platform dependent configuration
 
-build:ci-linux-common --copt='-Werror'
-build:ci-linux-common --copt='-Wno-error=#warnings'
-build:ci-linux-common --copt='-Wno-error=deprecated-declarations'
+build:ci-Linux-common --copt='-Werror'
+build:ci-Linux-common --copt='-Wno-error=#warnings'
+build:ci-Linux-common --copt='-Wno-error=deprecated-declarations'
 # keep in sync with .github/workflows/test.yml
-build:ci-linux-common --action_env=CC=/usr/lib/llvm-16/bin/clang --action_env=CXX=/usr/lib/llvm-16/bin/clang++
-build:ci-linux-common --host_action_env=CC=/usr/lib/llvm-16/bin/clang --host_action_env=CXX=/usr/lib/llvm-16/bin/clang++
+build:ci-Linux-common --action_env=CC=/usr/lib/llvm-16/bin/clang --action_env=CXX=/usr/lib/llvm-16/bin/clang++
+build:ci-Linux-common --host_action_env=CC=/usr/lib/llvm-16/bin/clang --host_action_env=CXX=/usr/lib/llvm-16/bin/clang++
 
-build:ci-linux --config=ci-linux-common
+build:ci-Linux --config=ci-Linux-common
 # Some tests (like Python import tests) take a long time to run and don't benefit much
 # from multi-platform testing. For that reason, we only run them in a single configuration
 # to minimize effect on CI pipeline runtime.
-build:ci-linux --build_tag_filters=-off-by-default
-test:ci-linux --test_tag_filters=-off-by-default --test_size_filters=small,medium,large,enormous
+build:ci-Linux --build_tag_filters=-off-by-default
+test:ci-Linux --test_tag_filters=-off-by-default --test_size_filters=small,medium,large,enormous
 
-build:ci-linux-debug --config=ci-linux-common --config=ci-limit-storage
-build:ci-linux-debug --config=debug --config=rust-debug
+build:ci-Linux-debug --config=ci-Linux-common --config=ci-limit-storage
+build:ci-Linux-debug --config=debug --config=rust-debug
 
-build:ci-linux-asan --config=ci-linux-common --config=ci-limit-storage
+build:ci-Linux-asan --config=ci-Linux-common --config=ci-limit-storage
 # we're really struggling to fit asan build into worker disk size
 # having asan without symbols is better than none
-build:ci-linux-asan --config=asan --copt="-g0" --strip=always
+build:ci-Linux-asan --config=asan --copt="-g0" --strip=always
 
+# Enable lld identical code folding to significantly reduce binary size.
```

Comment:
ICF is primarily useful for release builds – link time slightly more than doubles with this. You could make a case for it based on disk space usage and the time it takes to fetch binaries from cache, but based on the download speed that CI has this would be a net loss for test build time.